src/engine.rs

```diff
@@ -101,9 +104,15 @@ impl Reedline {
             need_full_repaint: false,
             partial_command: None,
             vi_engine: ViEngine::new(),
+            highlighter: Box::new(DefaultHighlighter::default()),
         }
     }
 
+    pub fn with_highlighter(mut self, external_commands: Vec<String>) -> Reedline {
```

Comment:
Nice! One small tweak: instead of taking in the external_commands here, take in the whole Highlighter.

Then, when you build a Reedline, you have more control over what Highlighter is used.

Reply:
Thank you;  initially i wanted to do it too but I had some compilation errors; if you can, try to propose your change, thanks.

Reply:
@mzanrosso - not sure how best to propose changes, but I did a branch based on yours: https://github.com/jonathandturner/reedline/tree/feature/buffer-colors

maybe you can compare the two and see the few spots I changed?

Reply:
That's okay, I usually meant a diff or a git prepare-patch but no problem, i will compare it asap.

Reply:
@jonathandturner  
I have implemented your changes, is there still something we can improve?